### PR TITLE
Add upgrade guide for v3.4

### PR DESCRIPTION
**Pull request by waiting-for-dev**
## Summary

We also add a helper to display the minimal requirements in terms of Ruby & Rails.

screenshot-localhost_3000-2023 04 20-15_23_59

- Closes https://github.com/solidusio/solidus/issues/5029

## Checklist

- [x] I have followed the [Diátaxis](https://diataxis.fr/) framework in my PR.
- [x] I have verified that the preview environment works correctly.

### CODE REVIEW

**Comment by waiting-for-dev** [comment]
Closes https://github.com/solidusio/solidus/issues/5029

File: docs/upgrading-solidus/v3.4.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+sidebar_position: 1
+---
+
+import PRLink from '@site/src/theme/PRLink';
+import MinimalRequirements from '@site/src/theme/MinimalRequirements';
+
+# Solidus v3.4 (2023-04-TBD)
+
+<MinimalRequirements ruby="2.7" rails="6.0" />
```
Comment (kennyadsl) [inline docs/upgrading-solidus/v3.4.mdx:10]: Nice! What do you think about adding this to the other upgrade guide as well?

File: src/theme/MinimalRequirements.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <div class="minimal_requirements">
+      <p>Minimal requirements:</p>
+      <dl>
+        <dt>
+          <img src="/img/ruby.png" alt="Ruby" />
+        </dt>
+        <dd>v{ props.ruby }</dd>
+        <dt>
+          <img src="/img/rails.png" alt="Rails" />
```
Comment (elia) [inline src/theme/MinimalRequirements.js:15]: SVGs would be better here, some links below:

https://commons.wikimedia.org/wiki/File:Ruby_logo.svg
https://rubyonrails.org/assets/images/logo.svg
https://commons.wikimedia.org/wiki/File:Ruby_On_Rails_Logo.svg
https://pressupinc.com/wp-content/uploads/2014/11/Ruby_on_Rails.svg_.png

**Review by elia — APPROVED** [review]
📈

File: docs/upgrading-solidus/v3.4.mdx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+This is a slimmer release, but a very important one as it'll be the last on the 3.x series. [Following our policy](https://solidus.io/release_policy/#last_minor_before_major), there won't be other changes except removing deprecated code before we release v4. That means if you adapt your store to be free of warnings, you should be ready to go when v4 is out (and it's just around the corner!).
+
+Please, remember to review our generic [upgrade guides](/getting-started/upgrading-solidus.mdx). You can also check the complete [Changelog](https://github.com/solidusio/solidus/blob/v3.4/CHANGELOG.md) in our repository.
```
Comment (kennyadsl) [inline docs/upgrading-solidus/v3.4.mdx:14]: I'm always afraid that people won't read it accurately and miss running the `solidus:update` task. What about adding it anyway as some form of TLDR? I'm sure they will go in the guide to better understand how to use the output of that task.

**Review by kennyadsl — APPROVED** [review]
Left a couple of non-blocking notes, great work!

File: docs/upgrading-solidus/v3.4.mdx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+---
+
+import PRLink from '@site/src/theme/PRLink';
+import MinimalRequirements from '@site/src/theme/MinimalRequirements';
+
+# Solidus v3.4 (2023-04-TBD)
+
+<MinimalRequirements ruby="2.7" rails="6.0" />
+
+This is a slimmer release, but a very important one as it'll be the last on the 3.x series. [Following our policy](https://solidus.io/release_policy/#last_minor_before_major), there won't be other changes except removing deprecated code before we release v4. That means if you adapt your store to be free of warnings, you should be ready to go when v4 is out (and it's just around the corner!).
```
Comment (RyanofWoods) [inline docs/upgrading-solidus/v3.4.mdx:12]: Do you mean `there won't be other changes except removing deprecated code in release v4`? @waiting-for-dev 🤔

File: docs/upgrading-solidus/v3.4.mdx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+Please, remember to review our generic [upgrade guides](/getting-started/upgrading-solidus.mdx). You can also check the complete [Changelog](https://github.com/solidusio/solidus/blob/v3.4/CHANGELOG.md) in our repository.
+
+## <PRLink description="New taxon and taxonomy validations" number="4851" />
+
+New validations were added to the `Taxon` and `Taxonomy` models but are behind the new temporary `extra_taxon_validations` and `extra_taxonomy_validations` preferences.
+
+- Taxonomy must have a unique name.
+- Taxon must have a unique name under the same parent Taxon (or at root level).
+- A validation was added in Taxon to ensure Taxonomies can only have one root.
```
Comment (RyanofWoods) [inline docs/upgrading-solidus/v3.4.mdx:22]: can we say "root Taxon" for clarity?

File: src/theme/MinimalRequirements.js (hunk 2 above)
Comment (waiting-for-dev) [reply src/theme/MinimalRequirements.js]: Good call. Updated.

File: docs/upgrading-solidus/v3.4.mdx (hunk 3 above)
Comment (waiting-for-dev) [reply docs/upgrading-solidus/v3.4.mdx]: I think you're right. I added the explicit task that needs to be run. Let me know what you think.

File: docs/upgrading-solidus/v3.4.mdx (hunk 4 above)
Comment (waiting-for-dev) [reply docs/upgrading-solidus/v3.4.mdx]: That's clearer. Updated!

File: docs/upgrading-solidus/v3.4.mdx (hunk 5 above)
Comment (waiting-for-dev) [reply docs/upgrading-solidus/v3.4.mdx]: :heavy_check_mark:

File: docs/upgrading-solidus/v3.4.mdx (hunk 1 above)
Comment (kennyadsl) [reply docs/upgrading-solidus/v3.4.mdx:10]: Definitely!

**Pull request merged**